File: ccc21j2.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
	ios_base::sync_with_stdio(false);
	cin.tie(NULL);

	int n;
	cin >> n;

	string winner;
	int highestBid = -1;
	for (int i = 0; i < n; ++i) {
		string name;
		int bid;
		cin >> name >> bid;
		
		if (highestBid < bid) {
			highestBid = bid;
			winner = name;
		}
	}
	
	cout << winner << "\n";
}

File: ccc21j2.java
import java.util.*;
import java.io.*;

public class ccc21j2 {
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

		int N = Integer.parseInt(br.readLine());
		String winnerName = "";
		int highestBid = -1;

		for (int i = 0; i < N; i++) {
			String name = br.readLine();
			int bid = Integer.parseInt(br.readLine());
			
			if (highestBid < bid) {
				highestBid = bid;
				winnerName = name;
			}
		}

		System.out.println(winnerName);
	}
}